### PR TITLE
profiles: Add m68k musl profile support

### PR DESCRIPTION
**Pull request by immolo**
sys-libs/musl: m68k keywording

After testing this profile and musl support has been working on my m68k system so is ready for review. I've added sys-libs/musl keywording at the same time as per the bug request.

Closes: https://bugs.gentoo.org/909040

### CODE REVIEW

File: profiles/default/linux/m68k/17.0/musl/make.defaults [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ARCH="m68k"
+ACCEPT_KEYWORDS="m68k ~m68k"
+
+ABI="m68k"
+DEFAULT_ABI="m68k"
+MULTILIB_ABIS="m68k"
+LIBDIR_m68k="lib"
+
+# Disable sandbox on this architecture
+FEATURES="-sandbox"
```
Comment (chewi) [inline profiles/default/linux/m68k/17.0/musl/make.defaults:16]: Why? It works fine for me. Not under qemu-user, but that's a special case. Also if the reason sandbox doesn't work is due to seccomp (think so, can't remember), then that was implemented for m68k [very recently](https://github.com/seccomp/libseccomp/pull/397).

**Review by chewi — CHANGES_REQUESTED** [review]

File: profiles/default/linux/m68k/17.0/musl/make.defaults [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2022-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+CHOST="m68k-unknown-linux-musl"
+CHOST_m68k="${CHOST}"
```
[the remark this answers is not in the document]
Comment (immolo) [reply profiles/default/linux/m68k/17.0/musl/make.defaults:5]: You are likely right on this as I added it to try an solve an issue and forgot to revert it.

File: profiles/default/linux/m68k/17.0/musl/make.defaults (hunk 1 above)
Comment (immolo) [reply profiles/default/linux/m68k/17.0/musl/make.defaults:16]: I shamelessly stole it from [here](https://github.com/gentoo/gentoo/blob/master/profiles/arch/m68k/make.defaults#L23)  the logic behind it being I'll work with what I knew worked then test and remove afterwards. With this information though I'll remove it now.

**Comment by chewi** [comment]
Please can you tidy up the history as we don't squash these commits. Also you deleted .editorconfig.

**Pull request closed without merging**